.gradient/prepare-datasets.sh

```diff
@@ -33,24 +11,17 @@ then
 fi
 
 echo "Starting preparation of datasets"
-# symlink exe_cache files
-exe_cache_source_dir="${PUBLIC_DATASET_DIR}/poplar-executables-pytorch-3-1-1"
-symlink-public-resources "${exe_cache_source_dir}" $POPLAR_EXECUTABLE_CACHE_DIR
-# Symlink squad
-symlink-public-resources "${PUBLIC_DATASET_DIR}/squad" "${HF_DATASETS_CACHE}/squad"
-# Symlink OGB Wiki dataset and checkpoint
-symlink-public-resources "${PUBLIC_DATASET_DIR}/ogbl_wikikg2_custom" "${DATASET_DIR}/ogbl_wikikg2_custom"
-
-# symlink local dataset used by vit-model-training notebook
-symlink-public-resources "${PUBLIC_DATASET_DIR}/chest-xray-nihcc-3" "${DATASET_DIR}/chest-xray-nihcc-3"
+cd "$(dirname "$0")"
+python -u ./symlink_datasets_and_caches.py
+cd -
 
 # pre-install the correct version of optimum for this release
 python -m pip install "optimum-graphcore>=0.5, <0.6"
 
 echo "Finished running setup.sh."
 # Run automated test if specified
-if [[ "$1" == "test" ]]; then
+if [[ "${1:-}" == "test" ]]; then
     bash /notebooks/.gradient/automated-test.sh "${@:2}"
```

Comment:
Unless `automated-test.sh` lacks an initial interpreter line (e.g. `#! /usr/bin/env bash` or similar) or isn't executable then we should invoke it directly, rather than forcing `bash`, e.g. simply:

```bash
/notebooks/.gradient/automated-test.sh "${@:2}"
```

... or, it's possible (although potentially unsafe, if the script uses clashing variables or itself exits) to execute the script in the context of this shell by invoking it as:

```bash
. /notebooks/.gradient/automated-test.sh "${@:2}"
```

(... with `source` being a synonym for `.`) - but this also means that we're using _this_ interpreter rather than the one the invoked script denotes.